### PR TITLE
Add old application name to /v1/register-yourself

### LINKED ISSUE #313: Add old application name to /v1/register-yourself
Root issue : https://github.com/openBackhaul/RegistryOffice/issues/310

Once a new application gets approved by TAR , RO have to send embed-yourself to the new application.
In the request-body of embed-yourself  , RO have to send the following details , 
- old-release-protocol
- old-release-address
- old-release-port

If there is a change in the application-name of an application , then RO cannot able to find the correct old-release information from it's config file.
image

**Proposal :**
One possible solution could be including an attribute "old-application-name" and "old-release-number" to the request body of the following services
- /v1/register-yourself - In [ApplicationPattern OAS](https://github.com/openBackhaul/ApplicationPattern/blob/develop/ApplicationPattern.yaml) - required parameter
- /v1/register-application - In [RegistryOffice OAS](https://github.com/openBackhaul/RegistryOffice/blob/develop/spec/RegistryOffice%2Bforwardings.yaml) - optional parameter (can be avoided if http-server/application-name == OldRelease/http-client/application-name are same. 

Process explanation:
1. /v1/register-yourself is triggered with data of {old-application-name and old-release-number}.
   This data can be configured into HttpClient instance of OldRelease and can be further used for /v1/register-application. 
2. As a callback of /v1/register-yourself,,,, /v1/register-application will be triggered.
 In the request-body of register-application, we can include the attributes {old-application-name and old-release-number}
3. When RO receives /v1/register-application, along with creating new instances for new-application-name and new-release-number, we can create a mapping for HttpClient instance UUID of {old-application-name and old-release-number} to HttpClient instance UUID of  {new-application-name and new-release-number} can be stored in **appData** of RO. 

@PrathibaJee , thanks for finding out the issue.
Kindly let us know your views.
Note : creating this issue in RegistryOffice, once the proposal is accepted, we shall raise an issue in ApplicationPattern for updating specification of /v1/register-yourself in OAS.

### LINKED ISSUE #314: To include a response-profile instance for /v1/inform-about-release-history-in-generic-representation to consider version 2.0.0
In the Config file , need to include a response profile instance for "/v1/inform-about-release-history-in-generic-representation" to include the version 2("local-id": "1") details as well. 

```
{
                    "uuid": "ro-2-0-0-response-p-016",
                    "profile-name": "response-profile-1-0:PROFILE_NAME_TYPE_GENERIC_RESPONSE_PROFILE",
                    "response-profile-1-0:response-profile-pac": {
                        "response-profile-capability": {
                            "operation-name": "/v1/inform-about-release-history-in-generic-representation",
                            "field-name": {
                                "field-name-reference": "/core-model-1-4:control-construct/logical-termination-point=ro-2-0-0-http-s-000/layer-protocol=0/http-server-interface-1-0:http-server-interface-pac/http-server-interface-capability/release-list=1/release-number"
                            },
                            "description": "Represents the first entry in the list of releases",
                            "datatype": "string"
                        },
                        "response-profile-configuration": {
                            "value": {
                                "value-reference": "/core-model-1-4:control-construct/logical-termination-point=ro-2-0-0-http-s-000/layer-protocol=0/http-server-interface-1-0:http-server-interface-pac/http-server-interface-capability/release-list=1/changes"
                            }
                        }
                    }
}
```

Kindly please change the sequence number of the profile uuid if it is not wished.

### PR DESCRIPTION
**Pull request by openBackhaul**
Please, regard second part in [AP pull-request#484](https://github.com/openBackhaul/ApplicationPattern/pull/484).

Fixes #313
Fixes #314

### CODE REVIEW

File: spec/RegistryOffice.yaml [hunk 1]
```diff
@@ -1193,6 +1193,17 @@ paths:
                   description: >
                     'Release of application that wants to register
                     find or create [/core-model-1-4:control-construct/logical-termination-point=ro-2-0-0-http-c-*/layer-protocol=0/http-client-interface-1-0:http-client-interface-pac/http-client-interface-configuration/release-number]'
+                preceding-application-name:
+                  type: string
```
Comment (PrathibaJee) [inline spec/RegistryOffice.yaml:1197]: shall we add the property "minLength: 3" for this attribute.

**Review by PrathibaJee — CHANGES_REQUESTED** [review]
Shall we also add the new attributes "preceding-application-name" , "preceding-release-number" to the callback "PromptForBequeathingDataCausesTransferOfListOfAlreadyRegisteredApplications". Thanks.

File: spec/RegistryOffice.yaml (hunk 1 above)
Comment (openBackhaul) [reply spec/RegistryOffice.yaml:1197]: Changed as proposed.

File: spec/RegistryOffice.yaml [hunk 2]
```diff
@@ -634,6 +634,16 @@ paths:
                           description: >
                             'Release of application that wants to register
                             from [/core-model-1-4:control-construct/logical-termination-point=ro-2-0-0-http-c-*/layer-protocol=0/http-client-interface-1-0:http-client-interface-pac/http-client-interface-configuration/release-number]'
+                        preceding-application-name:
```
Comment (PrathibaJee) [inline spec/RegistryOffice.yaml:637]: Thanks for including these attributes.
These attributes are not added to the examples. But that is not a problem as its **optional**.

**Review by PrathibaJee — APPROVED** [review]
Changes are as expected. Many thanks.

**Pull request merged**